### PR TITLE
Issue 3267 fix assertj deprecation

### LINKED ISSUE #3267: KubernetesResourceUtilTest : Replace AssertJ's deprecated `asList()` DSL method with `asInstanceOf(InstanceOfAssertFactories.list(type.class))`
### Component

JKube Kit

### Description
In the latest versions of Assertj `asList()` DSL method has been marked as deprecated:

https://github.com/eclipse-jkube/jkube/blob/d000d3ffdc1a18047df64f772391d8829f5b3cda/jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/api/util/KubernetesResourceUtilTest.java#L133

users are advised to use `asInstanceOf(InstanceOfAssertFactories.list(type.class))` instead, we need to update `asList()` calls with this. 

### Expected Behavior
AssertJ's `asList()` is replaced with `asInstanceOf(InstanceOfAssertFactories.list(type.class))`

### Acceptance Criteria
- [ ] `asList()` DSL method call is replaced with  `asInstanceOf(InstanceOfAssertFactories.list(type.class))` in all places

### PR DESCRIPTION
**Pull request by ShivamChavan01**
## Description

Fixes #3267  
This PR replaces the deprecated AssertJ `asList()` DSL method with `asInstanceOf(InstanceOfAssertFactories.list(type.class))` in the `KubernetesResourceUtilTest` class to ensure compatibility with the latest AssertJ versions.

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3280 (_2024-07-26T19:31:51Z_)

:gear: [JKube E2E Tests (10116648078)](https://github.com/eclipse-jkube/ci/actions/runs/10116648078) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3280)](https://github.com/eclipse-jkube/ci/actions/runs/10116648078/job/27979929771)

File: jkube-kit/enricher/specific/src/main/java/org/eclipse/jkube/kit/enricher/specific/KarafHealthCheckEnricher.java [hunk 1]
```diff
@@ -13,16 +13,17 @@
  */
 package org.eclipse.jkube.kit.enricher.specific;
 
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.eclipse.jkube.kit.common.Configs;
+import org.eclipse.jkube.kit.enricher.api.JKubeEnricherContext;
+
```
Comment (rohanKanojia) [inline jkube-kit/enricher/specific/src/main/java/org/eclipse/jkube/kit/enricher/specific/KarafHealthCheckEnricher.java:22]: How are these changes related to #3267?

File: jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/api/util/KubernetesResourceUtilTest.java [hunk 2]
```diff
@@ -13,6 +13,36 @@
...
+import static org.eclipse.jkube.kit.enricher.api.util.KubernetesResourceUtil.isContainerImage;
+import static org.eclipse.jkube.kit.enricher.api.util.KubernetesResourceUtil.mergeResources;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import static org.mockito.Mockito.spy;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
+
```
Comment (rohanKanojia) [inline jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/api/util/KubernetesResourceUtilTest.java:45]: could you please revert these import changes?

File: jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/api/util/KubernetesResourceUtilTest.java [hunk 3]
```diff
@@ -298,7 +300,7 @@ void mergeResources_whenDeploymentProvidedAndFirstDeploymentWithEmptySpec_thenSh
           .extracting(DeploymentSpec::getTemplate)
           .extracting(PodTemplateSpec::getSpec)
           .extracting(PodSpec::getContainers)
-          .asList()
+          .asInstanceOf(InstanceOfAssertFactories.list(Type.class))
```
Comment (rohanKanojia) [inline jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/api/util/KubernetesResourceUtilTest.java:303]: `list(..)` argument should be object that's composing previous list used in `extracting` i.e `getContainers`

File: jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/api/util/KubernetesResourceUtilTest.java (hunk 2 above)
Comment (ShivamChavan01) [reply jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/api/util/KubernetesResourceUtilTest.java:45]: yes  ill do it rn sorry for the inconvince
